dbt/adapters/snowflake/impl.py

```diff
@@ -419,3 +423,21 @@ def valid_incremental_strategies(self):
     def debug_query(self):
         """Override for DebugTask method"""
         self.execute("select 1 as id")
+
+    def get_adapter_run_info(cls, config: RelationConfig) -> AdapterTrackingRelationInfo:
```

Comment:
See my comment in the associated PR on `dbt-adapters`. I think this could be simplified to:

```python
@classmethod
def _get_adapter_run_info(cls, config) -> Dict[str, Any]:
    if (
        config
        and hasattr(config, "_extra")
        and (relation_format := config._extra.get("table_format"))
    ):
        return relation_format
    return None
```
Two more things, the `@classmethod` decorator is missing, and I didn't realize you could use the walrus operator in a compound `if` statement like that. I've found a few scenarios where I wanted to do that (like this scenario, when checking if the attribute exists first) and I either couldn't get it to work, or just assumed it didn't. That's nice to know!

Reply:
I didn't think I needed the decorator here. But I'll add it for continuity.